types.ts

```diff
@@ -349,57 +526,129 @@ export type VariousSupportedAuthentication =
+ *
+ * A column format is just a wrapper around a formula defined in the {@link formulas} section
+ * of your pack definition. It tells Coda to execute that particular formula using the value
+ * of the cell as input.
+ *
+ * The formula referenced by a Format must have exactly one required parameter.
+ *
+ * You may optionally specify one or more {@link matchers}, which are regular expressions
+ * that can be matched against values that users paste into table cells, to determine if
+ * this Format is applicable to that value. This is a discovery mechanism to help users
+ * realize that there is a pack that may augment their experience of working with such values.
+ *
+ * For example, if you're building a Wikipedia pack, you may write a matcher regular expression
+ * that looks for Wikipedia article URLs, if you have a formula that can fetched structured data
+ * given an article URL. This would help users discover that there is a pack that can fetch
+ * structured data given only a url.
+ *
+ * At present, matchers will only be run on URLs and not other text values.
+ */
 export interface Format {
+  /**
+   * The name of this column format. This will show to users in the column type chooser.
+   */
   name: string;
+  /** @deprecated Namespaces are being removed from the product. */
   formulaNamespace: string;
+  /**
+   * The name of the formula to invoke for values in columns using this format.
+   * This must correspond to the name of a formula defined in this pack.
+   */
   formulaName: string;
   /** @deprecated No longer needed, will be inferred from the referenced formula. */
   hasNoConnection?: boolean;
+  /**
+   * A brief, optional explanation of how users should use this format, for example, what kinds
+   * of values they should put in columns using this format.
+   */
   instructions?: string;
+  /**
+   * Any regular expressions that match URLs that the formula implementing this format
+   * is capable of handling. As described in {@link Format}, this is a discovery mechanism.
+   */
   matchers?: string[];
+  /**
+   * @deprecated Currently unused.
+   */
   placeholder?: string;
 }
 
+/**
+ * @ignore
+ */
 export enum FeatureSet {
   Basic = 'Basic',
   Pro = 'Pro',
   Team = 'Team',
   Enterprise = 'Enterprise',
 }
 
+/**
+ * @ignore
+ */
 export enum QuotaLimitType {
   Action = 'Action',
   Getter = 'Getter',
   Sync = 'Sync',
   Metadata = 'Metadata',
 }
 
+/**
+ * @ignore
+ */
 export enum SyncInterval {
   Manual = 'Manual',
   Daily = 'Daily',
   Hourly = 'Hourly',
   EveryTenMinutes = 'EveryTenMinutes',
 }
 
+/**
+ * @ignore
+ */
 export interface SyncQuota {
   maximumInterval?: SyncInterval;
   maximumRowCount?: number;
 }
 
+/**
+ * @ignore
+ */
 export interface Quota {
   monthlyLimits?: Partial<{[quotaLimitType in QuotaLimitType]: number}>;
   // TODO(alexd): Deprecate
   maximumSyncInterval?: SyncInterval;
   sync?: SyncQuota;
 }
 
+/**
+ * @deprecated Define these in the pack management UI instead.
+ *
```

Comment:
Should these also include `@ignore`? Or is that duplicative?

Reply:
I kept them non-ignored but deprecated with the thinking that these are indeed concepts that are an active part of building a pack, they've just moved from the code to the pack management UI. They'll still show up in TypeScript even if we hide them from TypeDoc, so I wanted the explanation to be readily viewable to a user who stumbles across these fields.